lightningd/lightningd.c

```diff
@@ -629,6 +630,11 @@ int main(int argc, char *argv[])
 	struct timers *timers;
 	const char *stop_response;
 	struct htlc_in_map *unprocessed_htlcs;
+	struct rlimit nofile = {1024, 1024};
+
+	/*~ Make sure that we limit ourselves to something reasonable. Modesty
+	 *  is a virtue. */
+	setrlimit(RLIMIT_NOFILE, &nofile);
```

Comment:
Would it be moderately useful to have this as a configurable option?  Or do we need to do this `setrlimit` ASAP before any command-line processing?

Reply:
No we can make it configurable, but for now this limits us to ~1000 peers, which should be an ok guess for now.

For non-root users we're unlikely to even have a higher limit since it requires system configuration changes. If this becomes an issue we can definitely introduce the option.